packages/connect-wallet-modal/src/components/WalletsModal/components/ConnectWalletModal/ConnectWalletModal.tsx

```diff
@@ -74,16 +48,52 @@ export const ConnectWalletModal = ({
     setShowOtherWallets((value) => !value);
   }, []);
 
-  const isWalletsListEmpty = walletsList.length === 0;
-  const isWalletsToggleButtonShown =
-    walletsListFull.length > walletsList.length || isShownOtherWallets;
+  const [walletsList, setWalletsList] = useState<WalletAdapterData[]>([]);
+
+  const isWalletsToggleButtonShown = useRef(false);
+
+  useEffect(() => {
+    // Asynchronously filling wallets list because there is an async wallet detection during wallets sorting.
+    // Actually, almost all wallets can be detected synchronously, so this process is expected to be fast,
+    // and a loading indicator is not required for now.
+    void (async () => {
+      const walletsListFull = await sortWalletsList({
```

Comment:
It looks like sortWalletsList will be called even in cases when it is not needed ("walletDataList, walletsShown, walletsPinned," does not change, but "inputValue" does")
I think it’s better to divide this logic into separate parts - call sortWalletsList separately

Reply:
Done